### PR TITLE
Implemented Automate Plagiarism / Copy Detection

### PR DESCRIPTION
**Pull request by DevIos01**
As we talked about in #1177 I have implemented the automation workflow to check through all existing games in sprig and compare them with the new PR file to check if it matches or not to improve our PR process.

Here is an example:
image

image

### CODE REVIEW

**Comment by DevIos01** [comment]
yo @SamDev-7 , totally, idk why i did not come up with that , will make few changes to this and some other thoughts after the PR

File: .github/scripts/plagiarismCheck.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { fileURLToPath } from 'url';
+
+const preprocessCode = (code) => {
+    code = code.replace(/\/\*[\s\S]*?\*\/|\/\/.*/g, '');
+    code = code.replace(/map`[\s\S]*?`/g, '');
+    code = code.replace(/bitmap`[\s\S]*?`/g, '');
+    return code;
+};
+
+const calculateSimilarity = (code1, code2) => {
```
[the remark this answers is not in the document]
Comment (grymmy) [reply .github/scripts/plagiarismCheck.js:12]: We could use something straight-forward like https://www.npmjs.com/package/fast-diff

**Review by LucasHT22 — CHANGES_REQUESTED** [review]
I trust you but why prettier?

**Comment by DevIos01** [comment]
> I trust you but why prettier?

Good Catch, this was used for the last few commit, but making a new commit with it being removed as we don't need it anymore :)

**Review by LucasHT22 — APPROVED** [review]
Looks everything ok to me, lmk when you ok with merge

**Comment by DevIos01** [comment]
Alight ready to rumble, open the door ;) 🕺 
giphy

**Pull request merged**